File: Yash/C++/DSA/4. Stacks/basicStacks.cpp
#include <iostream>
#include <stack>
using namespace std;

int main()
{
    stack <int> stack;
    stack.push(10);
    stack.push(20);
    stack.push(30);
    stack.push(40);
    stack.push(50);
    int a = 60;
    stack.push(60);

    while(!stack.empty())
    {
        cout<< stack.top()<<" ";
        stack.pop();
    }

}

File: Yash/Java/6. Stacks/BasicStacks.java
package stacks;
import java.util.Stack;
import java.util.Scanner;

public class BasicStacks {
	public static void main(String[] args)
	{
		Scanner s = new Scanner(System.in);
		Stack<Integer> st = new Stack<>();
		//isEmpty
		System.out.println("Is Empty : "+st.isEmpty());
		//push
		st.push(23);
		st.push(5);
		st.push(4);
		st.push(34); //top-most-elem
		System.out.println("Is Empty : "+st.isEmpty());
		//peek
		System.out.println("Top element : "+st.peek());
		System.out.println("Stack : "+st); //will print the whole stack
		System.out.println("Size : "+st.size());
//		st.pop(); // will remove the top elem
//		System.out.println(st);
//		System.out.println("Size : "+st.size());
//		st.pop();
//		System.out.println(st);
//		System.out.println("Size : "+st.size());
		
		System.out.println("Enter the index of element to be displayed : ");
		int ind = s.nextInt();
		System.out.println(st.elementAt(ind));
		
		
	}
}
